Model/Applications/PipedAPI.swift

```diff
@@ -491,6 +491,34 @@ final class PipedAPI: Service, ObservableObject, VideosAPI {
         )
     }
 
+    static func nonProxiedAsset(instance: Instance, asset: AVURLAsset) -> AVURLAsset? {
+        guard let instanceURLComponents = URLComponents(url: instance.apiURL, resolvingAgainstBaseURL: false),
+              var urlComponents = URLComponents(url: asset.url, resolvingAgainstBaseURL: false),
+              let hostItem = urlComponents.queryItems?.first(where: { $0.name == "host" }),
+              let hostValue = hostItem.value
+        else {
+            return nil
+        }
+
+        var instanceDomainAndTLD: String? {
+            return instanceURLComponents.host?.components(separatedBy: ".").suffix(2).joined(separator: ".")
+        }
+
+        var urlDomainAndTLD: String? {
+            return urlComponents.host?.components(separatedBy: ".").suffix(2).joined(separator: ".")
+        }
+
+        if instanceDomainAndTLD == urlDomainAndTLD {
+            urlComponents.host = hostValue
+        }
```

Comment:
@arekf I'm not sure about this part because here I assume that the proxy-domain is the same as the instance domain. Maybe I should just do:

```suggestion
        urlComponents.host = hostValue
```

What do you think?